### PR TITLE
Test

### PR DESCRIPTION
**Pull request by ali-aliabadi**
App class have basic changes.
PageParser class have basic changes.
Link consumer  Class have basic changes.
get domain method in link consumer changed.

### CODE REVIEW

File: crawler/src/main/java/ir/jimbo/crawler/App.java [hunk 1]
```diff
@@ -171,7 +171,7 @@ private static void consumerThreadInterruption(CountDownLatch countDownLatch, in
         LOGGER.info("end interrupting consumer threads");
     }
 
-    private static void initializeConfigurations(String[] args) {
+    static void initializeConfigurations(String[] args) throws IOException {
```
Comment (seyedalireza) [inline crawler/src/main/java/ir/jimbo/crawler/App.java]: write access level for this fucntion

File: crawler/src/main/java/ir/jimbo/crawler/LinkConsumer.java [hunk 2]
```diff
@@ -127,17 +133,34 @@ private void sendUriToKafka(String uri, Producer<Long, String> producer) {
...
+        try {
+            return !cacheService.isDomainExist(getDomain(uri));
+        } catch (NoDomainFoundException e) {
+            backToKafka = false;
+            return false;
+        }
     }
 
-    private String getDomain(String url) {
+    String getDomain(String url) {
```
Comment (seyedalireza) [inline crawler/src/main/java/ir/jimbo/crawler/LinkConsumer.java]: add access level for this function

File: crawler/src/main/java/ir/jimbo/crawler/App.java [hunk 3]
```diff
@@ -21,11 +21,11 @@
 
     private static final Logger LOGGER = LogManager.getLogger(App.class);
     private static ArrayBlockingQueue<String> queue;
-    private static RedisConfiguration redisConfiguration;
-    private static KafkaConfiguration kafkaConfiguration;
-    private static AppConfiguration appConfiguration;
-    private static LinkConsumer[] consumers;
-    private static PageParserThread[] producers;
+    static RedisConfiguration redisConfiguration;
```
Comment (seyedalireza) [inline crawler/src/main/java/ir/jimbo/crawler/App.java]: add access level

File: crawler/src/main/java/ir/jimbo/crawler/App.java [hunk 4]
```diff
@@ -95,17 +95,17 @@ private static void aliveThreadCounter(MetricConfiguration metrics) {
...
             if (consumer.isAlive()) {
                 counter.inc();
             }
         }
         return counter.getCount();
     }
 
-    private static long getAllWakeProducers(Counter counter) {
-        for (PageParserThread producer: producers) {
+    static long getAllWakeProducers(Counter counter) {
```
Comment (seyedalireza) [inline crawler/src/main/java/ir/jimbo/crawler/App.java]: add access level

File: crawler/src/main/java/ir/jimbo/crawler/App.java (hunk 1 above)
Comment (ali-aliabadi) [reply crawler/src/main/java/ir/jimbo/crawler/App.java]: intellij say package private is better

File: crawler/src/main/java/ir/jimbo/crawler/App.java (hunk 3 above)
Comment (ali-aliabadi) [reply crawler/src/main/java/ir/jimbo/crawler/App.java]: access set to private and setter and getter added

**Pull request merged**